test/run_test.py

```diff
@@ -1074,22 +1182,55 @@ def main():
         # downloading test cases configuration to local environment
         get_test_case_configs(dirpath=test_directory)
 
-    has_failed = False
     failure_messages = []
+
+    selected_tests_parallel = [x for x in selected_tests if not must_serial(x)]
+    selected_tests_serial = [x for x in selected_tests if x not in selected_tests_parallel]
+    print_to_stderr("parallel tests:\n {}".format("\n ".join(selected_tests_parallel)))
+    print_to_stderr("serial tests:\n {}".format("\n ".join(selected_tests_serial)))
+
+    proc_limit = 3
+    pool = mp.get_context("spawn").Pool(proc_limit, maxtasksperchild=1)
+    os.makedirs(REPO_ROOT / "test" / "test-reports", exist_ok=True)
+
+    def success_callback(res):
+        test, err_message, log_file_path = res
+        print_log_file(test, log_file_path)
+        if err_message is None:
+            return True
+        failure_messages.append(err_message)
+        print_to_stderr(err_message)
+        if not options.continue_through_error:
+            pool.terminate()
+        return False
+
     try:
-        for test in selected_tests:
+        os.environ['PARALLEL_TESTING'] = '1'
```

Comment:
Nit: do you want to make this an CI env option so when PARALLEL_TESTING is set to true, use multiprocessing?  It could be done in a separate PR

Reply:
this env var is checked later on to set the amount of memory per process for cuda

as for turning off/on parallel testing, i think thats a good idea, but im also not sure if anyone actually runs test/run_tests.py other than in CI